unstructured/ingest/v2/pipeline/steps/upload.py

```diff
@@ -44,18 +44,12 @@ def __post_init__(self):
     def process_whole(self, iterable: iterable_input):
         self.run(contents=iterable)
 
-    async def _process_async(self, iterable: iterable_input):
-        return await asyncio.gather(*[self.run_async(**i) for i in iterable])
-
-    def process_async(self, iterable: iterable_input):
-        logger.info("processing content async")
-        return asyncio.run(self._process_async(iterable=iterable))
-
+    @timed
     def __call__(self, iterable: iterable_input):
         logger.info(
             f"Calling {self.__class__.__name__} " f"with {len(iterable)} docs",  # type: ignore
         )
-        if self.process.is_async():
+        if self.process.is_async() and self.process.is_async():
```

Comment:
This seems strange. why test for is_async twice?

Reply:
Good catch, must have been a copy-paste error.